### ISSUE TITLE
Rule blocked: CORERULES-246

### ISSUE THREAD

**Issue by mhungria** [cdisc-org/cdisc-rules-engine#702]
- Attach test dataset files that can be used to test the new feature
- Replace the "###" in the issue title with the primary JIRA ticket number for the rule. 
- Fill in the following information

**Links to related JIRA Tickets**
- https://jira.cdisc.org/projects/CORERULES/issues/CORERULES-
- https://jira.cdisc.org/projects/CORERULES/issues/CORERULES-

**Rule Information**
- **Standard**:
- **Rule ID**:
- **Rule Description**:

**Describe the problem**
A clear and concise description of the problem.

**Describe the solution**
1) Need functionality to compare 2 numeric vectors.
2) Need functionality to populate a vector based on suffix on strings.

**Proposed rule logic**
```yaml
Replace this with an example of how the new solution could be written as a rule
```

**Screenshots**
If applicable, add screenshots to help explain your problem and solution.

**Comment by chowsanthony** [comment]
Note the possibility of zero-padded sequence number. Excerpt from section 4.4.11, SDTMIG v3.4:
> For example, if the type of disease milestone is "EPISODE OF DISEASE UNDER STUDY", the values of MIDS for instances of this type of event could include "EPISODE1", "EPISODE2"; or "EPISODE01", "EPISODE02", and so on.

**Comment by chowsanthony** [comment]
Also, the stripped sequence number is expected to start with 1. Excerpt from Domain SM assumption 2(b):
> Sequence numbers should start with 1 and indicate the chronological order of the instances of this type of disease milestone.

**Comment by mhungria** [comment]
> Note the possibility of zero-padded sequence number. Excerpt from section 4.4.11, SDTMIG v3.4:
> 
> > For example, if the type of disease milestone is "EPISODE OF DISEASE UNDER STUDY", the values of MIDS for instances of this type of event could include "EPISODE1", "EPISODE2"; or "EPISODE01", "EPISODE02", and so on.

Yes, in this case the suffix is any text after "EPISODE", including a missing suffix ...
We'll need the suffix as a required parameter to such a function call.

**Comment by SFJohnson24** [comment]
@mhungria it appears your original ticket is blank-- we would like to address this ticket but there is no description about what is needed for CG0546 besides the comments above.  Could you maybe add some details as to what is needed on the development side for you to proceed?

**Comment by mhungria** [comment]
@SFJohnson24, it is very strange! I'll look into it.

**Comment by eljanssens** [comment]
@mhungria to allow is to further progress toward v1.0, could you please check and provide additional details for this ticket?

**Comment by eljanssens** [comment]
@mhungria can you please add additional information on what is blocking you from programming CG0546 so that we can help you?

**Comment by mhungria** [comment]
Hi team, I added an update to the ticket description.

**Comment by gerrycampion** [comment]
I don't think we need the $records_per_usubjid_per_midstype record count operation check.
I think we need a `matches_regex` check to ensure that mids ends with numeric values.

Additionally, because there is the possibility of no padding even if MIDS goes past 9, I think we will need to modify the `target_is_sorted_by` operator, something like:
```yaml
operator: target_is_sorted_by
regex: ".*(\\d+$)"
```
When regex is provided, it will sort only on the first regex-captured group of the string. Before sorting, it should convert to numeric if possible.

I'm not sure if 0-padded numbers will be converted properly to numeric. If not, the regex may need to be modified to ignore the zero padding.
